File: sample/venomtyme.cs
/*
* Once upon a time...
*/

public class Cat
{
  public string Location { get; private set; }
  public int BirthDate { get; private set; }
  public int CurrentDate { get; private set; }
  public string[] Catlikes { get; private set; }

  public Cat(string location, int birthDate, int currentDate, string[] catlikes)
  {
    Location = location;
    BirthDate = birthDate;
    CurrentDate = currentDate;
    Catlikes = catlikes;
  }

  public int Age()
  {
    return calcAge();
  }

  private int calcAge()
  {
    return CurrentDate - BirthDate;
  }
}
class Program
{
  static void Main(string[] args)
  {
    // ...there was a cat named Katlyn

    var cat = new Cat(
      "Cataliputra",
      2016,
      2022,
      new string[] { "Tasty Food", "Long Naps" }
    );
  }
}

File: sample/venomtyme.php
<?php
/*
 * Once upon a time...
 */
class Cat {
  public string $location;
  public int $birthDate;
  public int $currentDate;
  public array $catlikes;

  public function __construct(array $props) {
    $this->location = $props['location'];
    $this->birthDate = $props['birthDate'];
    $this->currentDate = $props['currentDate'];
    $this->catlikes = $props['catlikes'];
  }

  public function age(): int {
    return $this->calcAge();
  }

  private function calcAge(): int {
    return $this->currentDate - $this->birthDate;
  }
}

// ...there was a cat named Katlyn
$Catppuccin = new Cat([
  'location' => 'Cataliputra',
  'birthDate' => 2016,
  'currentDate' => 2022,
  'catlikes' => ['Tasty Food', 'Long Naps']
]);
?>
